src/web3-utils.js

```diff
@@ -125,6 +128,54 @@ export async function getIsContractAccount(web3, account) {
   }
 }
 
+const gasPriceApi = 'https://ethgasstation.info/json/ethgasAPI.json'
+export async function getGasPrice({
+  mainnet: { safeMinimum = '3', disableEstimate } = {},
+} = {}) {
+  if (network.type !== 'main') {
+    // Hardcode 10 for non-mainnet networks
+    return toWei('10', 'gwei')
+  }
+
+  // Mainnet and gas estimation not disabled
+  if (!disableEstimate) {
+    const safeMinimumInWei = toWei(safeMinimum, 'gwei')
+    let priceInWei
+
+    try {
+      const response = await fetch(gasPriceApi, {
+        method: 'GET',
+        mode: 'cors',
+        cache: 'no-cache',
+      })
+      const jsonResponse = await response.json()
+      // Note that all prices from ethgasstation need to be divided by 10 to be in gwei.
+      // The response contains a list of suggested gas prices from 2-120 in
+      // gasPriceRange, so 40 is "slightly higher than the recommended price".
+      const fasterPrice = (jsonResponse.gasPriceRange[40] || 0) / 10
+      // Just in case, if this isn't available or is way too high,
+      // prefer the suggested safe low price.
+      const safePrice = (jsonResponse.safeLow || 0) / 10
+      const recommendedPrice =
+        fasterPrice < safePrice || fasterPrice - safePrice > 10
+          ? safePrice
+          : fasterPrice
+      priceInWei = toWei(recommendedPrice.toString(), 'gwei')
+    } catch (e) {
+      log('Error fetching gas price: ', e)
+    }
+
+    // If we couldn't find the price or it was lower than the safe minimum,
+    // use the safe minimum
+    priceInWei = new BN(priceInWei || 0).lt(new BN(safeMinimumInWei))
+      ? safeMinimumInWei
+      : priceInWei
+    return priceInWei
+  }
+
+  // Return undefined to let the web3 provider handle gas price
```

Comment:
Just a style nitpick, but we could return `undefined` early if `disableEstimate` is `true`, to avoid the large condition block.

Reply:
Fixed in 3ca7524.